apps/aecore/src/aec_block_key_candidate.erl

```diff
@@ -7,71 +7,104 @@
     N = Consensus:keyblocks_for_target_calc() + 1,
     case aec_blocks:height(Block) < N of
         true  ->
-            int_create_(Height, BlockHash, Block, Beneficiary, Miner, [], Protocol);
+            int_create_(Height, BlockHash, Block, Beneficiary, Miner, [], Protocol, Flags);
         false ->
             case N of
                 1 ->
-                    int_create_(Height, BlockHash, Block, Beneficiary, Miner, [], Protocol);
+                    int_create_(Height, BlockHash, Block, Beneficiary, Miner, [], Protocol, Flags);
                 _ ->
                     case aec_chain:get_n_generation_headers_backwards_from_hash(BlockHash, N) of
                         {ok, Headers} ->
-                            int_create_(Height, BlockHash, Block, Beneficiary, Miner, Headers, Protocol);
+                            int_create_(Height, BlockHash, Block, Beneficiary, Miner, Headers, Protocol, Flags);
                         error ->
                             {error, headers_for_target_adjustment_not_found}
                     end
             end
     end.
 
-int_create_(Height, PrevBlockHash, PrevBlock, Beneficiary, Miner, AdjChain, Protocol) ->
+int_create_(Height, PrevBlockHash, PrevBlock, Beneficiary, Miner, AdjChain, Protocol, Flags) ->
     {ok, Trees} = aec_chain_state:calculate_state_for_new_keyblock(Height, PrevBlockHash,
                                                                    Miner, Beneficiary, Protocol),
-    Block = int_create_block(Height, PrevBlockHash, PrevBlock, Miner, Beneficiary, Trees, Protocol),
+    Block = int_create_block(Height, PrevBlockHash, PrevBlock, Miner, Beneficiary, Trees, Protocol, Flags),
     Consensus = aec_blocks:consensus_module(Block),
     case Consensus:keyblock_create_adjust_target(Block, AdjChain) of
         {ok, AdjBlock} -> {ok, AdjBlock};
         {error, Reason} -> {error, {failed_to_adjust_target, Reason}}
     end.
 
-int_create_block(Height, PrevBlockHash, PrevBlock, Miner, Beneficiary, Trees, Protocol) ->
+int_create_block(Height, PrevBlockHash, PrevBlock, Miner, Beneficiary, Trees, Protocol, Flags) ->
     Consensus = aec_consensus:get_consensus_module_at_height(Height),
     PrevKeyHash = case aec_blocks:type(PrevBlock) of
                       micro -> aec_blocks:prev_key_hash(PrevBlock);
                       key   -> PrevBlockHash
                   end,
     Fork = aeu_env:get_env(aecore, fork, undefined),
-    InfoField = aec_chain_state:get_info_field(Height, Fork),
+    InfoField =
+        case maps:get(hc_hole, Flags, false) of
+            true ->
+                %% Keep backwards comatibility for ae
+                %% WIP Note: Not sure if this should live here.
+                case aec_chain_state:get_info_field(Height, Fork) of
+                    default when Protocol  >= ?MINERVA_PROTOCOL_VSN ->
+                    PointReleaseInfo = aeu_info:block_info(),
+                    Info = (1 bsl 31) bor PointReleaseInfo,
```

Comment:
I should create a hc_hole_flag function (or create_hc_hole_flag, and is_hc_hole functions) somewhere. (In aec_headers, perhaps).